controllers/cache_controller.go

```diff
@@ -322,8 +322,7 @@ func (r *cacheRequest) reconcileDataGrid(cacheExists bool, cache api.Cache) erro
 	spec := r.cache.Spec
 
 	if spec.Template == "" && spec.TemplateName == "" {
-		r.log.Info("ignoring cache as 'spec.template' and 'spec.TemplateName' are undefined")
-		return nil
+		return fmt.Errorf("unable to reconcile Cachhe as both 'spec.template' and 'spec.TemplateName' are undefined")
```

Comment:
`Cachhe` -> `Cache`

`spec.template` and `spec.TemplateName` should probably have `template` both either upper case or lower case. (Lower case is what is used in the yaml)